hack/lib/strimzi.bash

```diff
@@ -13,17 +13,128 @@ function install_strimzi {
+        replicas: 3
+        listeners:
+          - name: plain
+            port: 9092
+            type: internal
+            tls: false
+          - name: tls
+            port: 9093
+            type: internal
+            tls: true
+            authentication:
+              type: tls
+          - name: sasl
+            port: 9094
+            type: internal
+            tls: true
+            authentication:
+              type: scram-sha-512
+        config:
+          offsets.topic.replication.factor: 3
+          transaction.state.log.replication.factor: 3
+          transaction.state.log.min.isr: 2
+          log.message.format.version: "2.6"
+        storage:
+          type: jbod
+          volumes:
+          - id: 0
+            type: persistent-claim
+            size: 100Gi
+            deleteClaim: false
+      zookeeper:
+        replicas: 3
+        storage:
+          type: persistent-claim
+          size: 100Gi
+          deleteClaim: false
+      entityOperator:
+        topicOperator: {}
+        userOperator: {}
+EOF
 
   header "Waiting for Strimzi to become ready"
   oc wait kafka --all --timeout=-1s --for=condition=Ready -n kafka
+
+  header "Applying Strimzi TLS Admin user"
+  cat <<-EOF | oc apply -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaUser
+metadata:
+  name: my-tls-user
+  namespace: kafka
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+EOF
+
+  header "Applying Strimzi SASL Admin User"
+  cat <<-EOF | oc apply -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaUser
+metadata:
+  name: my-sasl-user
+  namespace: kafka
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: scram-sha-512
+EOF
+
+  header "Waiting for Strimzi admin users to become ready"
+  oc wait kafkauser --all --timeout=-1s --for=condition=Ready -n kafka
+
+  header "Deleting existing Kafka user secrets"
+  oc delete secret -n default my-tls-secret || true
+  oc delete secret -n default my-sasl-secret || true
+
+  header "Creating a Secret, containing TLS from Strimzi"
+  STRIMZI_CRT=$(oc -n kafka get secret my-cluster-cluster-ca-cert --template='{{index .data "ca.crt"}}' | base64 --decode )
+  TLSUSER_CRT=$(oc -n kafka get secret my-tls-user --template='{{index .data "user.crt"}}' | base64 --decode )
+  TLSUSER_KEY=$(oc -n kafka get secret my-tls-user --template='{{index .data "user.key"}}' | base64 --decode )
+
+  oc create secret --namespace default generic my-tls-secret \
+      --from-literal=ca.crt="$STRIMZI_CRT" \
+      --from-literal=user.crt="$TLSUSER_CRT" \
+      --from-literal=user.key="$TLSUSER_KEY"
+
+  header "Creating a Secret, containing SASL from Strimzi"
+  SASL_PASSWD=$(oc -n kafka get secret my-sasl-user --template='{{index .data "password"}}' | base64 --decode )
+  oc create secret --namespace default generic my-sasl-secret \
+      --from-literal=password="$SASL_PASSWD" \
+      --from-literal=user="my-sasl-user"
```

Comment:
your Strimzi has TLS enabled, and the user is `SCRAM-SHA-512`, this info is missing on the Secret here.

Hence you need something like:
https://github.com/knative-sandbox/eventing-kafka/blob/master/test/e2e-tests.sh#L369-L373

The channel defaults to `SASLTypePlaintext` - which is not working w/ the user you have here